server-common/src/test/java/org/apache/kafka/server/util/MockTime.java

```diff
@@ -27,15 +23,21 @@
  * 1. This has an associated scheduler instance for managing background tasks in a deterministic way.
  * 2. This doesn't support the `auto-tick` functionality as it interacts badly with the current implementation of `MockScheduler`.
  */
-class MockTime(currentTimeMs: Long, currentHiResTimeNs: Long) extends JMockTime(0, currentTimeMs, currentHiResTimeNs) {
-
-  def this() = this(System.currentTimeMillis(), System.nanoTime())
+public class MockTime extends org.apache.kafka.common.utils.MockTime {
```

Comment:
should this implement AutoCloseable so that we can shutdown the scheduler correctly?

Reply:
Hum... Implementing AutoCloseable alone does not help. We would need to change all the usages to call `close` to make it work, right?

My goal with this patch is to move it from core to server-common without changing it. I think that we could consider doing this later, if it brings anything.

Reply:
Many automatic bug finders will flag a class if it is marked as AutoCloseable but is not closed properly in the code. That is one advantage of using this interface that I find very useful. But yes, we can do it separately outside this PR.